src/cosmic_ray/interceptors/base.py

```diff
@@ -0,0 +1,31 @@
+from parso.tree import Node
+
+from cosmic_ray.operators.operator import Operator
+from cosmic_ray.work_db import WorkDB
+from cosmic_ray.work_item import WorkItem
+
+
+class Interceptor:
```

Comment:
The only justification I can find for this class is that you can later avoid re-parsing of code in your PragmaInterceptor. How much time does that actually take? I suspect its negligible, and this is class structure you've created is unnecessary complexity. Can't the pragma work be done in the framework of the original interceptor API?

I'm not opposed to API changes, but a) I'd like some discussion of them beforehand and b) they needs justification.

Reply:
With this, we can easier analyze the node of the work_item.
Even if we compute AST again, we have to find the corresponding node for a given work_item. For line-based interceptors, it can be easy but to find an expression or argument node, it's harder.

Reply:
> to find an expression or argument node, it's harder.
What kind of interceptor do you have in mind here?

Reply:
Ok, I see that your annotation interceptor makes a good case for needing access a this level. I'll need to think about this a bit.